packages/server/migrations/20230219162719_migrate_varchar_columns_to_text.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function (knex) {
+    await knex.raw(`
+        DO $bulkmigration$
+        BEGIN
+        EXECUTE (
+        SELECT
+            string_agg(
+                format (
+                'ALTER TABLE %I ALTER COLUMN %I TYPE text',
+                c.table_name,
+                c.column_name
+                ),
+            E';\n')
+        FROM information_schema.columns c
+            JOIN information_schema.tables t
+            ON t.table_schema = c.table_schema
+            AND t.table_name = c.table_name
+            AND t.table_type = 'BASE TABLE'
+        WHERE c.data_type = 'character varying'
+            AND c.table_schema NOT IN ('information_schema', 'pg_catalog')
+            AND c.table_name NOT IN ('migrations')
+        );
+        END
+        $bulkmigration$;
+    `);
+};
+
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+// eslint-disable-next-line no-unused-vars
+exports.down = function (knex) {
+    // no going back
```

Comment:
We should just be able to do `alterTable` in reverse for this migration in `down`

Reply:
Thanks for the suggestion. This approach was suggested by Ty.

If we allowed this migration to be undone, I'm concerned what would happen once there was data in the `cfda_list` column that exceeded 255 characters (see https://github.com/usdigitalresponse/usdr-gost/issues/633). Truncating the data in that column would have a bad effect.

Reply:
Yeah, reverting the migration in this case would be a destructive change. 

I'm generally wary of the notion of a "downgrade path" in these migration tools outside the context of testing – it generally seems better to determine an upgrade path given the specific nature of an observed problem.

Reply:
This makes sense to me - I'd imagine this is probably not something we'd want to run on prod directly, rather we would make another migration.